pkg/streamingpromql/operators/binary_operation_test.go

```diff
@@ -298,6 +299,231 @@ func TestBinaryOperation_SeriesMerging(t *testing.T) {
 			},
 			expectedError: `found duplicate series for the match group {env="test"} on the right side of the operation at timestamp 1970-01-01T00:00:00.002Z: {__name__="right_side", env="test", pod="g"} and {__name__="right_side", env="test", pod="e"}`,
 		},
+		"mixed float and histogram input series with no conflict": {
+			input: []types.InstantVectorSeriesData{
+				{
+					Floats: []promql.FPoint{
+						{T: 1, F: 10},
+						{T: 3, F: 30},
+					},
+				},
+				{
+					Histograms: []promql.HPoint{
+						{T: 2, H: &histogram.FloatHistogram{}},
+						{T: 4, H: &histogram.FloatHistogram{}},
```

Comment:
It'd be good to make these histograms slightly different (maybe set `Sum` or `Count`?) so we can confirm the correct values are returned